### PR TITLE
all brendon's requests done - including customname

### PR DESCRIPTION
**Pull request by sharmiamala**

### CODE REVIEW

File: CLI/event/cli_search.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                        help='Version of simulation groups. eg.18-9-3')
+    parser.add_argument('search', choices=[(i) for i in constants.SEARCH_OPTIONS],
+                        help='Search type. eg.event or site')
+    parser.add_argument('group', help='Simulation group. eg.v18p5')
+    parser.add_argument('-t', '--time', choices=[(i) for i in constants.TIME_OPTIONS],
+                        help='Time of occurrence. eg.historical or future')
+    parser.add_argument('-e', '--event_name', help='Event name. eg.Albury')
+    parser.add_argument('-s', '--single', nargs=2, help='latitude, longitude', metavar=('latitude', 'longitude'))
+    parser.add_argument('-m', '--multiple', help='path to a CSV file containing lat,lon')
+    parser.add_argument('-c', '--customname', help='customname for the location e.g:CBD')
```
Comment (sungeunbae) [inline CLI/event/cli_search.py]: I'm not sure how this will work if it is multiple and customname?

File: CLI/event/web_search.py [hunk 2]
```diff
@@ -139,88 +128,30 @@ def multiple():
...
-        print "The zip file is available here: ", socket.gethostname() + z
-    else:
-        print "WEB"
-        csvfile = os.path.join(app.config['TEMP_DIR'], csvfile)
-        locDict, coordsDict, distancesDict, within_domainDict = selection_utils.create_loc_dict(csvfile, ll_path, group, event, corners)
-        z = query.write_component_files(group, event, realisation, bin_path, locDict, coordsDict, TEMP_DIR, im_path,
-                                        distancesDict, acc_path, within_domainDict)
+    csvfile = os.path.join(app.config['TEMP_DIR'], csvfile)
+    z = webcli_common.get_multiple_result(group, event, realisation, csvfile)
+    if z is not "empty":
```
Comment (sungeunbae) [inline CLI/event/web_search.py]: Use None, not "empty"

File: CLI/event/web_search.py [hunk 3]
```diff
@@ -139,88 +128,30 @@ def multiple():
...
-        print "CLI"
-        print "The zip file is available here: ", socket.gethostname() + z
-        # return redirect(z)
-    else:
-        print "WEB"
-        print z
+@app.route('/single_result/<group>/<event>/<realisation>/<lat>/<lon>/<customname>')
+def single_result(group, event, realisation, lat, lon, customname):
+    z = webcli_common.get_single_result(group, event, realisation, lat, lon, customname)
+    if z is not "empty":
```
Comment (sungeunbae) [inline CLI/event/web_search.py]: same issue here. Use None

File: CLI/event/webcli_common.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    within_domainDict = {}
+    customname_dict = {}
+    is_within_domain = sqlite_utils.is_point_inside_domain(lat, lon, corners)
+    stationName, coords, distances = query.get_nearest_station(ll_path, float(lat), float(lon), group, event)
+    locDict[(lat, lon)] = stationName
+    coordsDict[(lat, lon)] = coords
+    distancesDict[(lat, lon)] = distances
+    within_domainDict[(lat, lon)] = is_within_domain
+    customname_dict[(lat, lon)] = customname
+    z = "empty"
```
Comment (sungeunbae) [inline CLI/event/webcli_common.py]: z=None

File: CLI/utils/selection_utils.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+
+# TODO: check this for valid input sequence
+def get_latlon(input):
+
+    latlon = raw_input("\nEnter the %s:"%input)
+    if(len(latlon) == 0):
+        print "%s should not be empty. Please enter a value" %(input)
+        latlon = get_latlon(input)
+
+    return latlon
```
Comment (sungeunbae) [inline CLI/utils/selection_utils.py]: What format do you expect? Leave a comment. Don't you check if the input is in the expected format?

File: CLI/utils/selection_utils.py [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+def get_customname():
+    customname = raw_input("\nEnter a customname for your location (optional), or press enter to proceed:")
+    return customname
```
Comment (sungeunbae) [inline CLI/utils/selection_utils.py]: I'm not sure how you can handle multiple locations?

File: CLI/utils/zipping_utils.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+                  'STATION LONGITUDE', 'DISTANCE', 'WITHIN DOMAIN', 'CUSTOM NAME'])
+
+    for lat, lon in locDict.keys():
+        stationName = locDict[(lat, lon)]
+        coord = coordsDict[(lat, lon)]
+        distance = distancesDict[(lat, lon)]
+        is_inside_domain = within_domainDict[(lat, lon)]
+        customname = customname_dict[(lat, lon)]
+        if customname is None:
+            customname = "None"
```
Comment (sungeunbae) [inline CLI/utils/zipping_utils.py]: Just asking - "None" vs "" which is better for customname?

**Comment by sungeunbae** [comment]
Another thing you should be adding. After user specified all the options through "guided" CLI, you should print the complete argument-based command that the user can copy and paste next time.

File: CLI/utils/zipping_utils.py (hunk 7 above)
Comment (sharmiamala) [reply CLI/utils/zipping_utils.py]: changed to ""

File: CLI/utils/selection_utils.py (hunk 6 above)
Comment (sharmiamala) [reply CLI/utils/selection_utils.py]: changed to display information and get the correct options

File: CLI/utils/selection_utils.py (hunk 5 above)
Comment (sharmiamala) [reply CLI/utils/selection_utils.py]: There is a validate function which takes care of the validation of lat, lon and customname already. Modified.

File: CLI/event/webcli_common.py (hunk 4 above)
Comment (sharmiamala) [reply CLI/event/webcli_common.py]: changed

File: CLI/event/web_search.py (hunk 3 above)
Comment (sharmiamala) [reply CLI/event/web_search.py]: changed

File: CLI/event/web_search.py (hunk 2 above)
Comment (sharmiamala) [reply CLI/event/web_search.py]: changed

File: CLI/event/cli_search.py (hunk 1 above)
Comment (sharmiamala) [reply CLI/event/cli_search.py]: changed to display information and get the correct options

File: CLI/event/cli_search.py [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+def get_event_search_result(selection_dict):
+    if (selection_dict['location'] == 'single'):
+        z = webcli_common.get_single_result(selection_dict['quake'], selection_dict['event'], selection_dict['realisation'],
+                                     selection_dict['latitude'], selection_dict['longitude'],
+                                     selection_dict['customname'])
+    else:
+        z = webcli_common.get_multiple_result(selection_dict['quake'], selection_dict['event'], selection_dict['realisation'],
+                                       selection_dict['csvfile'])
+
+    if z is not "empty":
```
[the remark this answers is not in the document]
Comment (sharmiamala) [reply CLI/event/cli_search.py]: changed

File: CLI/event/webcli_common.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+def get_multiple_result(group, event, realisation, csvfile):
+    # print "multiple_result()"
+    bin_path, im_path, ll_path, acc_path, corners, emprical_path, fault_list_path = query.get_paths_from_db(group,
+                                                                                                            event,
+                                                                                                            realisation)
+
+    loc_dict, coords_dict, distances_dict, within_domain_dict, customname_dict, is_within_domain = selection_utils.create_loc_dict(
+        csvfile, ll_path, group, event, corners)
+    z = "empty"
```
[the remark this answers is not in the document]
Comment (sharmiamala) [reply CLI/event/webcli_common.py]: changed

**Comment by sharmiamala** [comment]
All changes completed

File: CLI/utils/selection_utils.py [hunk 10]
```diff
@@ -9,7 +9,7 @@
 import re
 
 
-def validate_csv_file(user_input):
+def validate(user_input):
```
Comment (sungeunbae) [inline CLI/utils/selection_utils.py]: This should return either True or False. If there is an error, it should just print the error message. Returning a string "pass" for success is bad. This is the same thing as "empty" we discussed yesterday.

File: CLI/event/cli_search.py [hunk 11]
```diff
@@ -103,19 +120,40 @@ def get_site_search_result(selection_dict):
...
+
     if args.single:
         latitude = args.single[0]
         longitude = args.single[1]
         selection_dict["location"] = "single"
         selection_dict["latitude"] = latitude
         selection_dict["longitude"] = longitude
+        user_input_list = [[latitude, longitude, customname]]
+        is_input_valid = selection_utils.validate(user_input_list)
+        if is_input_valid != "pass":
```
Comment (sungeunbae) [inline CLI/event/cli_search.py]: No, it should be either True or False. Not a string "pass"

File: CLI/event/cli_search.py [hunk 12]
```diff
@@ -103,19 +120,40 @@ def get_site_search_result(selection_dict):
...
+            exit(1)
     elif args.multiple:
         csvfile = args.multiple
         selection_dict["location"] = "multiple"
         selection_dict["csvfile"] = csvfile
+        with open(csvfile, 'rb') as f:
+            reader = csv.reader(f)
+            user_input_list = list(reader)
+        is_input_valid = selection_utils.validate(user_input_list)
+        if is_input_valid != "pass":
```
Comment (sungeunbae) [inline CLI/event/cli_search.py]: same comment

File: CLI/utils/selection_utils.py [hunk 13]
```diff
@@ -245,7 +245,7 @@ def select_input_type(selection_dict):
             customname = None
 
         user_input_list = [[lat, lon, customname]]
-        is_input_valid = validate_csv_file(user_input_list)
+        is_input_valid = validate(user_input_list)
         if is_input_valid != "pass":
```
Comment (sungeunbae) [inline CLI/utils/selection_utils.py]: True or False

**Comment by sharmiamala** [comment]
All changes done

File: CLI/utils/selection_utils.py [hunk 14]
```diff
@@ -16,7 +16,7 @@ def validate(user_input):
     :return: "pass" if the input is valid, otherwise a different string containing an error
     """
     print "\n.............................................. Start of Validation ..................................."
-    message = "pass"
+    message = True
```
[the remark this answers is not in the document]
Comment (sharmiamala) [reply CLI/utils/selection_utils.py]: Changed accordingly

**Comment by sharmiamala** [comment]
validate function modified

File: CLI/utils/selection_utils.py [hunk 15]
```diff
@@ -44,13 +46,15 @@ def validate(user_input):
         else:
             message = msg + str(
                 lat) + "'. Please provide valid inputs. "
+            is_valid = False
             print message
```
Comment (sungeunbae) [inline CLI/utils/selection_utils.py]: "print message " is added so many times in the code. Is it necessary?
Why don't you just print message once at the end of the function when is_valid is False?

File: CLI/utils/selection_utils.py [hunk 16]
```diff
@@ -13,69 +13,69 @@ def validate(user_input):
...
         l = len(i)
-        if l < min_elements:  # if the columns are less than min_elements
+        if l < min_elements:
             message = "Input field should contain minimum "+ str(
-        min_elements) + " columns with numeric values. But the given input was " + str(
-                i) + "'. Please provide valid inputs. "
+        min_elements) + " columns with numeric values. But the given columns are less than " + str(
+                min_elements) + " in "+str(i)
             is_valid = False
             print message
```
Comment (sungeunbae) [inline CLI/utils/selection_utils.py]: Do you mean to leave this line?

**Comment by sharmiamala** [comment]
yes, because if it does not have enough columns then no point in proceeding
further with the validation. We exit. It needs to have atleast 2 numeric
columns with lat,lon

**Pull request merged**